test/e2e/all_in_one_test.go

```diff
@@ -89,6 +97,76 @@ const (
 	dblessURL  = "https://raw.githubusercontent.com/Kong/kubernetes-ingress-controller/%v.%v.x/deploy/single/all-in-one-dbless.yaml"
 )
 
+func TestDeployAllInOneDBLESSGateway(t *testing.T) {
+	t.Log("configuring all-in-one-dbless.yaml manifest test")
+	t.Parallel()
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	t.Log("building test cluster and environment")
+	addons := []clusters.Addon{}
+	addons = append(addons, metallb.New())
+	if b, err := loadimage.NewBuilder().WithImage(imageLoad); err == nil {
+		addons = append(addons, b.Build())
+	}
+	builder := environments.NewBuilder().WithAddons(addons...)
+	if clusterVersionStr != "" {
+		clusterVersion, err := semver.ParseTolerant(clusterVersionStr)
+		require.NoError(t, err)
+		builder.WithKubernetesVersion(clusterVersion)
+	}
+	env, err := builder.Build(ctx)
+	require.NoError(t, err)
+	defer func() {
+		assert.NoError(t, env.Cleanup(ctx))
+	}()
+
+	require.NoError(t, clusters.KustomizeDeployForCluster(ctx, env.Cluster(), "github.com/kubernetes-sigs/gateway-api/config/crd?ref=v0.4.0"))
+
+	t.Log("deploying kong components")
+	manifest, err := getTestManifest(t, dblessPath)
+	require.NoError(t, err)
+	deployment := deployKong(ctx, t, env, manifest)
+	for i, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == "ingress-controller" {
+			deployment.Spec.Template.Spec.Containers[i].Env = append(deployment.Spec.Template.Spec.Containers[i].Env,
+				corev1.EnvVar{Name: "CONTROLLER_FEATURE_GATES", Value: "Gateway=true"})
+		}
+	}
+
+	t.Log("deploying kong components")
+	forDeployment := metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("app=%s", deployment.Name),
+	}
+
+	// Annoyingly, metav1.Time.Now() in the test may desync from the time in-cluster, so we need to extract this
+	// directly from the Pod
+	podList, err := env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
+	require.NoError(t, err)
+	require.GreaterOrEqual(t, len(podList.Items), 1)
+	preUpdateTimestamp := podList.Items[0].Status.StartTime
+
+	_, err = env.Cluster().Client().AppsV1().Deployments(deployment.Namespace).Update(ctx,
+		deployment, metav1.UpdateOptions{})
+	require.NoError(t, err)
+	require.Eventually(t, func() bool {
+		podList, err := env.Cluster().Client().CoreV1().Pods(deployment.Namespace).List(ctx, forDeployment)
+		require.NoError(t, err)
+		for _, pod := range podList.Items {
+			if pod.Status.StartTime.Before(preUpdateTimestamp) || pod.Status.StartTime.Equal(preUpdateTimestamp) {
+				return false
+			}
+		}
+		return true
+	}, kongComponentWait, time.Second)
```

Comment:
I'm slightly confused what we're accomplishing here, can you help me to better understand?

Reply:
We're patching the deployment after creation in the test since there's no existing way to inject envvars into the manifests (and I didn't want to write a more generic Kustomize tool), and I wanted some way to ensure that the Pod is using the updated template.

This probably isn't strictly necessary since the HTTPRoute/Gateway verification after will imply the update succeeded, they'll just effectively have less time to complete. Do you think it makes sense to drop it for simplicity?

Reply:
830de2564174a652bb0aae9cb16033a049484985 just removes it, with no adverse effects in a test run.

Reply:
Yeah that makes sense :+1: